### ISSUE TITLE
Naming inconsistency

### ISSUE THREAD

**Issue by ogvalt** [catalyst-team/catalyst#770]
**Describe the bug**
I found that some names agruments in framework aren't consistent. 
So for example:
```
class SupervisedRunner(Runner):
    """Runner for experiments with supervised model."""

    _experiment_fn: Callable = SupervisedExperiment

    def __init__(
        self,
        model: Model = None,
        device: Device = None,
        input_key: Any = "features", 
        output_key: Any = "logits",
        input_target_key: str = "targets", # This argument corresponds to input_key argument in CriterionCallback
    ):

class CriterionCallback(_MetricCallback):
    """Callback for that measures loss with specified criterion."""

    def __init__(
        self,
        input_key: Union[str, List[str], Dict[str, str]] = "targets", # This argument corresponds to input_target_key argument in SupervisedRunner
        output_key: Union[str, List[str], Dict[str, str]] = "logits",
        prefix: str = "loss",
        criterion_key: str = None,
        multiplier: float = 1.0,
        **metric_kwargs,
    ):
```

**To Reproduce**
1. Check files: `catalyst.core.callback.metric.py` and `catalyst.dl.runner.supervised.py`

**Expected behavior**
I expect that names would be consistent across the framework and means the same

**Comment by Scitator** [comment]
Hi,
Could you please describe where do you see an inconsistency and why? It would be better, if you could propose your own examples, like `how it now` vs `how it should be`.

For now I see all names consistent, cause both `input_key` and  `input_target_key` are from `state.input`, while `output_key` is stored into `state.output`.

**Comment by ogvalt** [comment]
**How it now** 

**input_target_key** corresponds to **input_key** argument in **CriterionCallback**
        `input_target_key: str = "targets"` 

**input_key** corresponds to **input_target_key** argument in **SupervisedRunner**
        `input_key: Union[str, List[str], Dict[str, str]] = "targets"`

```
class SupervisedRunner(Runner):
    """Runner for experiments with supervised model."""

    _experiment_fn: Callable = SupervisedExperiment

    def __init__(
        self,
        model: Model = None,
        device: Device = None,
        input_key: Any = "features", 
        output_key: Any = "logits",
        # !!!input_target_key corresponds to input_key argument in CriterionCallback!!!
        input_target_key: str = "targets", 
    ):

class CriterionCallback(_MetricCallback):
    """Callback for that measures loss with specified criterion."""

    def __init__(
        self,
        # !!!input_key corresponds to input_target_key argument in SupervisedRunner!!!
        input_key: Union[str, List[str], Dict[str, str]] = "targets",         
        output_key: Union[str, List[str], Dict[str, str]] = "logits",
        prefix: str = "loss",
        criterion_key: str = None,
        multiplier: float = 1.0,
        **metric_kwargs,
    ):
```

**How it should**

Since `CriterionCallback` calcucates criterion between `target` (or `input_target_key`) and `output_key` it should be better to rename `input_key` to `input_target_key` as in the `SupervisedRunner`

```

class CriterionCallback(_MetricCallback):
    """Callback for that measures loss with specified criterion."""

    def __init__(
        self,
        # previous version
        input_key: Union[str, List[str], Dict[str, str]] = "targets", 
        # suggested version
        input_target_key: Union[str, List[str], Dict[str, str]] = "targets",         
        ...
    ):
```
**Also I suggest to check this across whole framework**

**Comment by Scitator** [comment]
Thank you for explanation :)
I this case, I think you are right then, it looks reasonable to share name between Runner and Callbacks. 

And what do you think about `output_key`? Should we rename it into something more accurate, like `output_logits_key`?

**Comment by ogvalt** [comment]
For me `output_key` was confusing at first. But after it's totally make sence from architectural point of view. If you're considering renaming I would rather suggest `output_prediction_key` as more descriptive term, or just leave `output_key`.

**Comment by Scitator** [comment]
👌 I agree, `output_prediction_key` looks much better.
What do your think, would you like to make a pull request with desired changes? I think, I would be much easier for newcomers to understand Callbacks & Metrics system then. Thank you!

**Comment by Scitator** [comment]
btw, @ogvalt speaking about naming, what if we change `SuppervisedRunner` keys a bit?
for example
```
class SupervisedRunner(Runner):
    def __init__(
        self,
        model: Model = None,
        device: Device = None,
        input_keys: Tuple[Any] = ("features", "targets"), 
        output_keys: Any = "logits",
    ):
```

I think, such refactoring could be easier to understand
extra: + you don't need to refactor callbacks system (which `input/output` keys logic looks okay with refactored `SupervisedRunner`)

**Comment by ogvalt** [comment]
@Scitator, looks better in terms of refactoring, but lacks of expressiveness. 
I would rather suggest to use dict, like this:
```
class SupervisedRunner(Runner):
    def __init__(
        self,
        model: Model = None,
        device: Device = None,
        input_keys: Dict[str, Any] = {"features": ..., "targets": ...}, 
        output_keys: Any = "logits",
    ):
```
it would be easier to distinguish features and targets, doensn't lack of expressiveness, but still looks better in terms of  refactoring.

**Comment by Scitator** [comment]
Looks good, but here is the question, based on input/output keys parsing
`input_keys` both `features` and `targets` could be `str`, `List/Tuple`, `Dict` (rare case) or even `None`.
In this case,
```
class SupervisedRunner(Runner):
    def __init__(
        self,
        model: Model = None,
        device: Device = None,
        input_keys: Dict[str, Any] = {"features": ('features1', 'features2', 'features3'), "targets": ('target1', 'target2', 'target3')}, 
        output_keys: Any = ("logits1", "logits2", "logits3"),
    ):
```
Looks a bit complicated, I think.

Moreover, `input_keys` are only needed to create a key-value storage from input batch correctly :)

**Comment by Scitator** [comment]
Dear @ogvalt ,
As a result of our discussion, what do you think about refactoring 
```
class SupervisedRunner(Runner):
    def __init__(
        self,
        model: Model = None,
        device: Device = None,
        input_keys: Tuple[Any] = ("features", "targets"), 
        output_keys: Any = "logits",
    ):
```
Would you have some time to make a pull request with this feature?

**Comment by Scitator** [comment]
@ogvalt How are you? what do you think about my proposal?

**Comment by ogvalt** [comment]
@Scitator Sorry for late reply, I had to much stuff going on.
So, if the goal was to create correct key-value storage from input batch then
```
...
input_keys: Tuple[Any] = ("features", "targets"), 
...
```
wouldn't fullfil this goal.
So, I'm not sure, that your proposal won't harm internal logic of framework

**Comment by Scitator** [comment]
@ogvalt What do you think it wouldn't fullfil the goal?

**Comment by Scitator** [comment]
nope, we need it
@ogvalt ping :)

**Comment by Scitator** [comment]
still relevant :)
